pss/pss_test.go

```diff
@@ -0,0 +1,73 @@
+package pss
+
+import (
+	"context"
+	"crypto/rand"
+	"io/ioutil"
+	"os"
+	"reflect"
+	"testing"
+
+	"github.com/ethersphere/swarm/chunk"
+	trojan "github.com/ethersphere/swarm/pss/trojan"
+	"github.com/ethersphere/swarm/storage/localstore"
+)
+
+func TestTrojanChunkRetrieval(t *testing.T) {
+	var err error
+	ctx := context.TODO()
+
+	localStore := newMockLocalStore(t)
+
+	testTargets := [][]byte{
+		{57, 120},
+		{209, 156},
+		{156, 38},
+		{89, 19},
+		{22, 129}}
+	payload := []byte("RECOVERY CHUNK")
+	topic := trojan.NewTopic("RECOVERY")
+
+	// call Send to store trojanChunk in store
+	var ch chunk.Chunk
+
+	pss := NewPss(localStore)
+
+	if ch, err = pss.Send(ctx, testTargets, topic, payload); err != nil {
+		t.Fatal(err)
+	}
+
+	// verify store, that trojan chunk has been stored correctly
+	var storedChunk chunk.Chunk
+	if storedChunk, err = localStore.Get(ctx, chunk.ModeGetRequest, ch.Address()); err != nil {
+		t.Fatal(err)
+	}
+
+	if !reflect.DeepEqual(ch, storedChunk) {
```

Comment:
well this smells. But ok for now. lets ask @janos what the correct test would be here.